examples/with-stripe-typescript/app/components/CheckoutForm.tsx

```diff
@@ -23,26 +35,57 @@ export default function CheckoutForm(): JSX.Element {
       [e.currentTarget.name]: e.currentTarget.value,
     });
 
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = async (
+    e
+  ): Promise<void> => {
+    e.preventDefault()
+
+    if (props.uiMode === 'hosted') {
+      const { url } = await createCheckoutSession(
+        new FormData(e.target as HTMLFormElement)
+      )
+
+      window.location.assign(url as string)
```

Comment:
Should this use Next.js `router` instead?

Reply:
That's only for navigating to routes within the Next.js app, right? This is an external URL.

Reply:
You can use external URLs as well.

* `redirect` works on the server and in Server Actions (ie. Server Actions)
* `router.push` works in client-side components
* `window.location.assign` also works in client-side components

Reply:
Ah, I was using `window.location` based on the note in your [documentation](https://nextjs.org/docs/pages/api-reference/functions/use-router#routerpush:~:text=You%20don%27t%20need,for%20those%20cases.).